src/tests/partial-length.cpp

```diff
@@ -0,0 +1,262 @@
+    assert_rnp_success(rnp_input_from_path(&input, "data/test_partial_length/message.txt.partial-1g"));
+    assert_rnp_success(rnp_output_to_null(&output));
+    rnp_op_verify_t verify = NULL;
+    assert_rnp_success(rnp_op_verify_create(&verify, ffi, input, output));
+    assert_rnp_success(rnp_op_verify_execute(verify));
+
+    // cleanup
+    assert_rnp_success(rnp_op_verify_destroy(verify));
+    assert_rnp_success(rnp_input_destroy(input));
+    assert_rnp_success(rnp_output_destroy(output));
+    assert_rnp_success(rnp_ffi_destroy(ffi));
+}
+
+TEST_F(rnp_tests, test_partial_length_first_packet_length)
+{
+    rnp_ffi_t           ffi = NULL;
+    rnp_input_t         input = NULL;
+    rnp_output_t        output = NULL;
+    rnp_op_sign_t       sign = NULL;
+    rnp_key_handle_t    key = NULL;
+
+    // uncacheable size will emulate unknown length from callback source
+    size_t uncacheable_size = PGP_INPUT_CACHE_SIZE + 1;
+
+    // init ffi
+    test_partial_length_init(&ffi, RNP_LOAD_SAVE_SECRET_KEYS);
+
+    // generate a sequence of octets with appropriate length using callback
+    dummy_reader_ctx_st reader_ctx;
+    reader_ctx.dummy = 'X';
+    reader_ctx.remaining = uncacheable_size;
+    assert_rnp_success(rnp_input_from_callback(&input, dummy_reader, NULL, &reader_ctx));
+
+    assert_rnp_success(rnp_output_to_memory(&output, uncacheable_size + 1024));
+
+    assert_rnp_success(rnp_op_sign_create(&sign, ffi, input, output));
+    assert_rnp_success(rnp_locate_key(ffi, "keyid", "7BC6709B15C23A4A", &key));
+    assert_rnp_success(rnp_op_sign_add_signature(sign, key, NULL));
+
+    assert_rnp_success(rnp_key_handle_destroy(key));
+    key = NULL;
+
+    // signing
+    assert_rnp_success(rnp_op_sign_execute(sign));
+
+    // read from the saved packets
+    pgp_source_t    src;
+    uint8_t         *mem = NULL;
+    size_t          len = 0;
+    assert_rnp_success(rnp_output_memory_get_buf(output, &mem, &len, false));
+    assert_rnp_success(init_mem_src(&src, mem, len, false));
+
+    // skip first packet (one-pass signature)
+    pgp_packet_body_t body;
+    assert_rnp_success(stream_read_packet_body(&src, &body));
+    assert_int_equal(body.tag, PGP_PTAG_CT_1_PASS_SIG);
+    free_packet_body(&body);
+
+    // checking next packet header (should be partial length literal data)
+    uint8_t flags = 0;
+    assert_int_equal(src_read(&src, &flags, 1), 1);
+    assert_int_equal(flags, PGP_PTAG_ALWAYS_SET | PGP_PTAG_NEW_FORMAT | PGP_PTAG_CT_LITDATA);
+    // checking length
+    bool last = true; // should be reset by stream_read_partial_chunk_len()
+    assert_true(stream_read_partial_chunk_len(&src, &last) >= PGP_PARTIAL_PKT_FIRST_PART_MIN_SIZE);
+    assert_false(last);
+
+    // cleanup
+    src_close(&src);
+    assert_rnp_success(rnp_op_sign_destroy(sign));
+    assert_rnp_success(rnp_input_destroy(input));
+    assert_rnp_success(rnp_output_destroy(output));
+    assert_rnp_success(rnp_ffi_destroy(ffi));
+}
```

Comment:
Empty line afterwards would save us from seeing 'No empty line at the end of file' warning.